src/components/marketplace/cancel_order_button.tsx

```diff
@@ -0,0 +1,68 @@
+import { SignedOrder } from '@0x/connect';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+
+import { cancelOrder } from '../../store/actions';
+import { UIOrder } from '../../util/types';
+
+interface OwnProps {
+    order: UIOrder;
+}
+
+interface DispatchProps {
+    onCancelOrder: (order: SignedOrder) => Promise<any>;
+}
+
+type Props = OwnProps & DispatchProps;
+
+interface State {
+    isLoading: boolean;
+}
+
+const BUTTON = styled.button`
+    background: none;
+    border: none;
+    color: #C4C4C4;
+    display: inline-block;
+    height: 1em;
+    outline: 0;
+
+    &:hover {
+        cursor: pointer;
+    }
+`;
+
+class CancelOrderButton extends React.Component<Props, State> {
```

Comment:
Just a thought: It feels weird to have a container component used inside a presentational component. But I can't think of a downside of doing this (the parent component is still testable, if I'm not missing anything), and the alternative is passing down a callback for the cancel order, which is cumbersome.

Reply:
Feels a little weird but it encapsulates what the button does. We can refactor the base code to have presentational containers and presentational components. I'm not 100% sure, but having the latters splitted out may make UI customization easier.

Reply:
Yeah, no need to change this, I'm OK with it.